tests/test_flow/test_protocols.py

```diff
@@ -28,44 +28,50 @@
 
 
 @pytest.mark.parametrize("value", PICKLABLE_VALUES)
-def test_picklable_value(builder, value):
+def test_picklable_value(builder, manager, value):
```

Comment:
It feels a little bit awkward to me that we need to use this `manager` object to create our counter objects, since the connection between a process manager and a counter is not very obvious. We could use a `make_counter` factory fixture to make the relationship clearer:

    @pytest.fixture
    def make_counter(manager):
        def _make_counter():
            return manager.ResettingCounter()

    def def test_picklable_value(builder, make_counter, value):
        counter = make_counter()

What do you think? Is this an improvement?

Reply:
I didn't realize you can easily define factory methods in `pytest` like this. This is pretty neat, thanks!